routers/user/oauth_userinfo.go

```diff
@@ -0,0 +1,29 @@
+package user
+
+import (
+	"fmt"
+
+	"code.gitea.io/gitea/modules/context"
+)
+
+// UserInfoResponse represents a successful userinfo response
+type UserInfoResponse struct {
+	Sub      string `json:"sub"`
+	Name     string `json:"name"`
+	Username string `json:"preffered_username"`
+	Email    string `json:"email"`
+	Picture  string `json:"picture"`
+}
+
+// UserInfoOAauth responds with OAuth formatted userinfo
+func UserInfoOAuth(ctx *context.Context) {
+	user := ctx.User
```

Comment:
Thanks for this PR!

There is a chance that user may be nil, could you do a check that user != nil to ensure a nil referenceerror doesn't happen?

Reply:
I think this may be more complicated. Not the nil check, but the correctness of the userinfo endpoint itself. AuthorizeOAuth reads the bearer token, and I suspect this code may need to do it too. I am relying on `reqSignIn`, but I didn't see any Authorization header parsing, so I suspect this may need a full rewrite.

Particularly, the `reqSignIn` requires a sign-in and should error out with a 403 before reaching this code.

Can you confirm that the required signed in context takes the Authorization header?

Reply:
userinfo endpoint should work exclusively with token provided as this is endpoint that is called as s2s request so there won't be user cookies etc for user session